enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -161,7 +161,7 @@ def _create_kernel_namespace(self, kernel_username, service_account_name):
         # the namespace's default ServiceAccount.  Since this is always done when creating a namespace, we can
         # delete the RoleBinding when deleting the namespace (no need to record that via another member variable).
 
-        namespace = kernel_username + '-' + self.kernel_id
+        namespace = re.sub('[^0-9a-z]+', '-', kernel_username).lower() + '-' + self.kernel_id
```

Comment:
I think the `.lower()` should be inside the substitution method so that username is lower-cased prior.  Otherwise a user name of `Foo.bar` will yield `-oo-bar` instead of `foo-bar`.

However, the kubernetes docs state that namespace names are a bit more restrictive.  From their error message text...
> a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character

So a KERNEL_USERNAME of "-foo" isn't valid.  As a result, I think we should just drop the offending characters, so "-Foo.bar-dud" will yield: "foobardud" via 
```python
re.sub('[^0-9a-z]+', '', kernel_username.lower())
```
Ideally, we should allow embedded hyphens to be preserved (e.g., `foobar-dud`) but I wasn't able to figure out an RE to do that.  If you can, that would be great but I wouldn't knock yourself out.

In my local testing, I needed to `import re`.  Did you not find this necessary?

Reply:
```
        namespace = re.sub('[^0-9a-z]+', '-', kernel_username.lower()) + '-' + self.kernel_id
        while namespace.startswith('-'):
            namespace = namespace[1:]
```
would remove heading -s and without deleting other -s.